service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/CaseSubmissionControllerSubmittedTest.java

```diff
@@ -74,88 +76,83 @@ void shouldReturnUnsuccessfulResponseWithMalformedData() {
 
     @Test
     void shouldBuildNotificationTemplatesWithCompleteValues() throws Exception {
-        Map<String, Object> completeHmctsParameters = getCompleteParameters()
-            .put("court", FAMILY_COURT)
-            .build();
+        SubmitCaseHmctsTemplate completeHmctsParameters = getCompleteParameters(new SubmitCaseHmctsTemplate());
+        completeHmctsParameters.setCourt(FAMILY_COURT);
 
-        Map<String, Object> completeCafcassParameters = getCompleteParameters()
-            .put("cafcass", CAFCASS_COURT)
-            .build();
+        SubmitCaseCafcassTemplate completeCafcassParameters = getCompleteParameters(new SubmitCaseCafcassTemplate());
+        completeCafcassParameters.setCafcass(CAFCASS_COURT);
 
         postSubmittedEvent("core-case-data-store-api/callback-request.json");
 
         verify(notificationClient).sendEmail(
             HMCTS_COURT_SUBMISSION_TEMPLATE,
             HMCTS_ADMIN_EMAIL,
-            completeHmctsParameters,
+            completeHmctsParameters.toMap(mapper),
             CASE_REFERENCE.toString());
 
         verify(notificationClient).sendEmail(
             CAFCASS_SUBMISSION_TEMPLATE,
             CAFCASS_EMAIL,
-            completeCafcassParameters,
+            completeCafcassParameters.toMap(mapper),
             CASE_REFERENCE.toString());
 
         verify(notificationClient, never()).sendEmail(
             HMCTS_COURT_SUBMISSION_TEMPLATE,
             "FamilyPublicLaw+ctsc@gmail.com",
-            completeHmctsParameters,
+            completeHmctsParameters.toMap(mapper),
```

Comment:
Shouldn't need the toMap part, sendEmail has an overloaded method that accepts the object which has a base type of `NotifyData` instead of the map, same in other places in this class as well